metrics/prometheus_test.go

```diff
@@ -8,108 +8,101 @@ import (
-	finish := time.Time{}.Add(5.0 * time.Second)
 
-	// Result 1: No successes, no failures, full run
-	successes := []run.ApplyAttempt{}
-	failures := []run.ApplyAttempt{}
-	fullRun := true
-	p.processResult(run.Result{Successes: successes, Failures: failures, FullRun: fullRun, Start: start, Finish: finish})
-	// Ensure count 1 for latency metric with full_run=true, success=true
-	l := makeLatencyRegexp(true, true, 1)
-	metricsRaw := requestContentBody(handler)
-	assert.True(l.MatchString(metricsRaw))
+	testCases := []testCase{
+		// Case 1: No successes, no failures, full run
+		{
+			[]run.ApplyAttempt{},
+			[]run.ApplyAttempt{},
+			run.FullRun,
+			[]string{
+				// Expect count 1 for latency metric with run_type=fullRun, success=true
+				makeLatencyPattern(run.FullRun, true, 1),
+			},
+		},
+		// Case 2: Successes, no failures, full run
+		{
+			[]run.ApplyAttempt{{FilePath: "file1"}, {FilePath: "file2"}},
+			[]run.ApplyAttempt{},
+			run.FullRun,
+			[]string{
+				// Expect count 2 for latency metric with run_type=fullRun, success=true
+				makeLatencyPattern(run.FullRun, true, 2),
+				// Expect count 1 for file1 with success=true
+				makeFilePattern("file1", true, 1),
+				// Expect count 1 for file2 with success=true
+				makeFilePattern("file2", true, 1),
+			},
+		},
+		// Case 3: Successes, failures, full run
+		{
+			[]run.ApplyAttempt{{FilePath: "file1"}, {FilePath: "file3"}},
+			[]run.ApplyAttempt{{FilePath: "file2"}},
+			run.FullRun,
+			[]string{
+				// Expect count 1 for latency metric with run_type=fullRun, success=false
+				makeLatencyPattern(run.FullRun, false, 1),
+				// Expect count 2 for file1 with success=true
+				makeFilePattern("file1", true, 2),
+				// Expect count 1 for file3 with success=true
+				makeFilePattern("file3", true, 1),
+				// Expect count 1 for file2 with success=false
+				makeFilePattern("file2", false, 1),
 
-	// Result 2: Successes, no failures, full run
-	successes = []run.ApplyAttempt{{FilePath: "file1"}, {FilePath: "file2"}}
-	failures = []run.ApplyAttempt{}
-	fullRun = true
-	p.processResult(run.Result{Successes: successes, Failures: failures, FullRun: fullRun, Start: start, Finish: finish})
-	// Ensure count 2 for latency metric with full_run=true, success=true
-	l = makeLatencyRegexp(true, true, 2)
-	metricsRaw = requestContentBody(handler)
-	assert.True(l.MatchString(metricsRaw))
-	// Ensure count 1 for file1 with success=true
-	f := makeFileRegexp("file1", true, 1)
-	assert.True(f.MatchString(metricsRaw))
-	// Ensure count 1 for file2 with success=true
-	f = makeFileRegexp("file2", true, 1)
-	assert.True(f.MatchString(metricsRaw))
+				// Ensure that previous metrics remain unchanged.
+				// Expect count 2 for latency metric with run_type=fullRun, success=true
+				makeLatencyPattern(run.FullRun, true, 2),
+				// Expect count 1 for file2 with success=true
+				makeFilePattern("file2", true, 1),
+			},
+		},
+		// Case 4: Successes, failures, quick run
+		{
```

Comment:
Unfortunately they are quite dependent at the moment, although this is by design. The test cases use the same file names to check that the values are updated properly (e.g. after first run, check the count is 1... after second run, check that the count is two... etc)